### DISCUSSION TITLE
Add Alpine.js Dispatch event to action button

### DISCUSSION THREAD

**Discussion by mfonabasiudobia** [Power-Components/livewire-powergrid#523] [Ideas]
In as much as we have action buttons for route, emit, emitTo, etc. 

I would love to get an action button that can dispatch a javascript event with parameters, I had a project that needed this functionality.

![Untitled](https://user-images.githubusercontent.com/31806957/180661409-f50762f9-95de-46c4-8f20-12c5f7018e9c.png)

**Comment by luanfreitasdev** [comment]
This is very interesting, currently it is possible to do it using [Action Rules](https://livewire-powergrid.com/#/table/action-rules?id=setattributestring-attribute-null-string-value-null).

**Reply by mfonabasiudobia** [reply]
What is available currently is livewire emit event, I think also having javascript dispatch event would be nice, mainly to show a popup or confirmation dialog when a button is clicked.

With livewire emit event, you have to wait for a request to be sent to the server before the action is executed, with the dispatch event, it will be done on the client side, mainly for popups and confirmation when a button is clicked.

I hope you get the idea now

**Reply by luanfreitasdev** [reply]
You can use setAttribute('x-on:click="$dispatch ......)

**Reply by mfonabasiudobia** [reply]
Nice!!!!!!!
